schematic/json_schema_functions.py

```diff
@@ -0,0 +1,368 @@
+        wiki_title: The title the created wiki will have
+        schema_version: The version the created JSON Schema will have
+
+    Returns:
+        The uri of the uploaded JSON Schema,
+        the Synapse id of the created entity view,
+        and the Synapse Wiki object
+    """
+    if not schema_name:
+        schema_name = f"{datatype}.schema"
+    if not entity_view_name:
+        entity_view_name = f"{datatype} entity view"
+    if not wiki_title:
+        wiki_title = f"{datatype} wiki"
+
+    json_schema_uri = create_and_bind_json_schema(
+        syn=syn,
+        data_model_path=data_model_path,
+        datatype=datatype,
+        synapse_org_name=synapse_org_name,
+        synapse_entity_id=synapse_entity_id,
+        schema_name=schema_name,
+        schema_version=schema_version,
+    )
+
+    entity_view_id = create_json_schema_entity_view(
+        syn=syn,
+        entity_id=synapse_entity_id,
+        parent_id=synapse_parent_id,
+        entity_view_name=entity_view_name,
+    )
+    wiki = create_entity_view_wiki(
+        syn=syn,
+        entity_view_id=entity_view_id,
+        owner_id=synapse_parent_id,
+        title=wiki_title,
+    )
+    return (json_schema_uri, entity_view_id, wiki)
+
+
+def create_and_bind_json_schema(  # pylint: disable=too-many-arguments
+    syn: Synapse,
+    data_model_path: str,
+    datatype: str,
+    synapse_org_name: str,
+    synapse_entity_id: str,
+    schema_name: Optional[str] = None,
+    schema_version: str = "0.0.1",
+) -> str:
+    """
+    This function:
+    - creates a json schema
+    - uploads it to Synapse
+    - binds it to a Synapse entity
+
+    Arguments:
+        syn: A Synapse object thats been logged in
+        data_model_path: A path to the data model use dot create the JSON Schema
+        datatype: The datatype in the data model to create the JSON Schema for
+        synapse_org_name: The Synapse org to upload the JSON Schema to
+        synapse_entity_id: The ID of the entity in Synapse to bind the JSON Schema to
+        schema_name: The name the created JSON Schema will have
+        schema_version: The version the created JSON Schema will have
+
+    Returns:
+        The uri of the uploaded JSON Schema
+    """
+    js_service = syn.service("json_schema")
+    dmge = create_dmge(data_model_path)
+
+    if not schema_name:
+        schema_name = f"{datatype}.schema"
+
+    json_schema = create_json_schema(
+        dmge=dmge,
+        datatype=datatype,
+        schema_name=schema_name,
+        write_schema=False,
+        use_property_display_names=False,
+    )
+    json_schema_uri = upload_json_schema(
+        syn=syn,
+        json_schema=json_schema,
+        synapse_org_name=synapse_org_name,
+        schema_name=schema_name,
+        schema_version=schema_version,
+    )
+
+    js_service.bind_json_schema(json_schema_uri, synapse_entity_id)
+    return json_schema_uri
+
+
+def upload_json_schema(
```

Comment:
TODO: when we get there, we need to work with the data managers around how best to version their data models and how that propagates back into Synapse